### ISSUE TITLE
[BUG] NullReferenceException reading V1.0 camel cased client-side encryption metadata

### ISSUE THREAD

**Issue by craigjensen** [Azure/azure-sdk-for-net#29750]
### Library name and version

Azure.Storage.Blobs 12.13.0

### Describe the bug

NullReferenceException when attempting to deserialize client side encryption metadata (encryptiondata) that was serialized with a previous version of the SDK configure to use camel casing (stack below).  The problem appears to be in the various EncryptionDataSerializer.ReadPropertyValue methods that use JsonProperty.NameEquals to do a case-sensitive name comparison.

"innermostType": System.NullReferenceException,
"innermostMessage": Object reference not set to an instance of an object.,
"details":    at Azure.Storage.Blobs.BlobClientSideDecryptor.GetAndValidateEncryptionDataOrDefault(IDictionary`2 metadata)
   at Azure.Storage.Blobs.Specialized.BlobBaseClient.<OpenReadInternal>d__93.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Azure.Storage.Blobs.Specialized.BlobBaseClient.<OpenReadAsync>d__91.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at System.Runtime.CompilerServices.ConfiguredTaskAwaitable`1.ConfiguredTaskAwaiter.GetResult()
   at Microsoft.Bot.ChannelConnector.TemporaryAttachmentStore.<>c__DisplayClass23_0.<<GetAttachmentContentAsync>b__0>d.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at System.Runtime.CompilerServices.ConfiguredTaskAwaitable`1.ConfiguredTaskAwaiter.GetResult()
   at Microsoft.Bot.ChannelConnector.TemporaryAttachmentStore.<AzureBlobOperationAsync>d__33`1.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at System.Runtime.CompilerServices.ConfiguredTaskAwaitable`1.ConfiguredTaskAwaiter.GetResult()
   at Microsoft.Bot.ChannelConnector.TemporaryAttachmentStore.<GetAttachmentContentAsync>d__23.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at System.Runtime.CompilerServices.ConfiguredTaskAwaitable`1.ConfiguredTaskAwaiter.GetResult()
   at Microsoft.Bot.ChannelConnector.AttachmentsController.<>c__DisplayClass3_0.<<GetAttachment>b__0>d.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter`1.GetResult()
   at Microsoft.Bot.ChannelConnector.WebApiControllerExtensions.<ExecuteAndRespond>d__0`1.MoveNext(),

### Expected behavior

Json deserialization should be case-insensitive to support reading existing metadata

### Actual behavior

NullReferenceException when attempting to read existing blobs that use client-side encryption and were written by a previous version of the SDK

### Reproduction Steps

1) set the default JsonSerializer to camelCase:
   JsonConvert.DefaultSettings = () => new JsonSerializerSettings()
   {
     ContractResolver = new CamelCasePropertyNamesContractResolver()
   };
2) Write a blob with client-side encryption enabled using V1 encryption with the previous SDK (Microsoft.Azure.Storage.Blob V11.2.3)
3) Verify that the encryption metadata Json is camelCased.  It should look like this:
{"encryptionMode":"FullBlob","wrappedContentKey":{"keyId":"key1","encryptedKey":"B79XoZ7Ky+/9xJps7SFFLZOedlSfuLP+mukLtuUKEpWNZHUR5CIbsQ==","algorithm":"A256KW"},"encryptionAgent":{"protocol":"1.0","encryptionAlgorithm":"AES_CBC_256"},"contentEncryptionIV":"/e8I2sr5ow1jtMw/oDJB3g==","keyWrappingMetadata":{"encryptionLibrary":".NET 11.2.3"}}
4) Attempt to read the blob with the new SDK (Azure.Storage.Blobs V12.13.0)

Another more simple repro is to write a blob using the 12.13.0 SDK and manually change the casing of the encryptionData to be camelCased and then attempt to read back the blob.

**Comment by SaurabhSharma-MSFT** [comment]
@craigjensen We are looking into it and get back to you for any additional information.

**Comment by SaurabhSharma-MSFT** [comment]
@craigjensen You may be getting this error as you need to re-encrypt previously encrypted data with client-side encryption v2 version.  Can you please try as per the directions over [here](https://docs.microsoft.com/en-us/azure/storage/blobs/client-side-encryption?tabs=dotnet#reencrypt-previously-encrypted-data-with-client-side-encryption-v2) and let me know if you still get this error?

**Comment by craigjensen** [comment]
We are trying to switch from V1 to V2 encryption.  The problem is that V2 can't read our existing V1 blobs due to the issue described above.

**Comment by boydc2014** [comment]
@SaurabhSharma-MSFT do we have any update on this one, we are blocked by this bug that new SDK can't read existing blob thus not able to start the migrate at all.

**Comment by jaschrep-msft** [comment]
Acknowledging bug. Apologies for delay. Will investigate.

**Comment by jaschrep-msft** [comment]
Diagnosing this as a bug in the legacy library. The older library should not have allowed the user to overwrite JSON property casing, as this creates an incompatible experience with the other legacy implementations it existed alongside.

While I see the "correct" solution to this to be a script that updates the metadata casing in the cloud before migrating, I understand the fear in tampering with this critical key for data retrieval using code that has not been as thoroughly tested as the implementations themselves. Updating to a case-insensitive deserialization makes sense to me.

**Comment by jaschrep-msft** [comment]
@craigjensen we have made a [patch release](https://www.nuget.org/packages/Azure.Storage.Blobs/12.13.1) with the above PR included. Please reopen this issue if the problem is not resolved, or file a separate issue if you notice any other issues in migrating to client-side encryption v2.
